packages/server-wallet/src/protocols/__test__/defunder.test.ts

```diff
@@ -0,0 +1,119 @@
+
+    let channel = await Channel.forId(testChannelObj.channelId, testKnex);
+    const defunder = new Defunder(store, chainService, logger);
+
+    testKnex.transaction(async tx => {
+      // Defunder should take no actions as there is no conclusion proof.
+      expect(await defunder.crank(channel, tx)).toEqual({
+        didSubmitTransaction: false,
+        isChannelDefunded: false,
+      });
+      expect(spy).not.toHaveBeenCalled();
+
+      // Defunder should submit a transaction since there is a conclusion proof.
+      const state = testChannelObj.wireState(5);
+      channel = await store.addSignedState(channel.channelId, state, tx);
+      expect(await defunder.crank(channel, tx)).toEqual({
+        didSubmitTransaction: true,
+        isChannelDefunded: false,
+      });
+      expect(spy).toHaveBeenCalledWith(channel.support);
+
+      // The channel has NOT been fully defunded.
+      // Defunder does not complete
+      await Funding.updateFunding(tx, channel.channelId, '0x03', testChannelObj.assetHolderAddress);
+      expect(await defunder.crank(channel, tx)).toEqual({
+        didSubmitTransaction: false,
+        isChannelDefunded: false,
+      });
+
+      // The channel has been fully defunded
+      // Defunder completes
+      await Funding.updateFunding(tx, channel.channelId, '0x00', testChannelObj.assetHolderAddress);
+      expect(await defunder.crank(channel, tx)).toEqual({
+        didSubmitTransaction: false,
+        isChannelDefunded: true,
+      });
+    });
+  });
+
+  it('Channel is defunded via pushOutcome transaction', async () => {
+    // Channel has yet to be concluded
+    await testChannelObj.insertInto(store, {
+      participant: 0,
+      states: [3, 4],
+      funds: 8,
+    });
+    const state3 = testChannelObj.signedStateWithHash(3);
+    const state4 = testChannelObj.signedStateWithHash(4);
+
+    const chainService = new MockChainService();
+    const spy = jest.spyOn(chainService, 'pushOutcomeAndWithdraw');
+    const logger = createLogger(defaultTestConfig());
+
+    const channel = await Channel.forId(testChannelObj.channelId, testKnex);
+    const defunder = new Defunder(store, chainService, logger);
+    await AdjudicatorStatusModel.insertAdjudicatorStatus(testKnex, channel.channelId, 1, [
+      state4,
+      state3,
+    ]);
+    await AdjudicatorStatusModel.setFinalized(testKnex, channel.channelId, 1, 1, 1);
+
+    await testKnex.transaction(async tx => {
+      expect(await defunder.crank(channel, tx)).toEqual({
+        didSubmitTransaction: true,
+        isChannelDefunded: false,
+      });
+      expect(spy).toHaveBeenCalledWith(state4, channel.myAddress);
+
+      // The channel has been fully defunded
+      // Defunder completes
+      await Funding.updateFunding(tx, channel.channelId, '0x00', testChannelObj.assetHolderAddress);
+      expect(await defunder.crank(channel, tx)).toEqual({
+        didSubmitTransaction: false,
```

Comment:
Why should this be `false`? It was just `true`. It now sounds a bit like the transaction was un-submitted.

Reply:
I think `didSumbitTransaction` is a bit confusing. Added a comment in 3dddf847f